core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -476,7 +476,7 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
   private def logInvalidStateTransitionAndReturnError(transactionalId: String,
                                                       transactionState: TransactionState,
                                                       transactionResult: TransactionResult) = {
-    debug(s"TransactionalId: $transactionalId's state is $transactionState, but received transaction " +
+    error(s"TransactionalId: $transactionalId's state is $transactionState, but received transaction " +
```

Comment:
Did we intend to keep this at error level?

Reply:
Yes, I found this error message is very useful when debugging.

Reply:
Stale requests are fenced before this point. But I can make it WARN.